### PR TITLE
Update fleet to v0.3.4-rc3

### PR DESCRIPTION
**Pull request by nickgerace**
Relevant issue: https://github.com/rancher/rancher/issues/30516

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/charts/commit/1dd374f9b1859f272cbcf87a64093824843a7355 @nickgerace Can you follow this PR and add patch? There is a bug in script that add packageVersion in wrong place so you have to put patch to remove rc version first

**Comment by nickgerace** [comment]
Fixed! @StrongMonkey

File: packages/fleet/generated-changes/patch/Chart.yaml.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- charts-original/Chart.yaml
++++ charts/Chart.yaml
+@@ -8,8 +8,8 @@
+   catalog.cattle.io/provides-gvr: clusters.fleet.cattle.io/v1alpha1
+   catalog.cattle.io/release-name: fleet
+ apiVersion: v2
+-appVersion: 0.3.4-rc3
++appVersion: 0.3.3
```
Comment (StrongMonkey) [inline packages/fleet/generated-changes/patch/Chart.yaml.patch:8]: Should be 0.3.4?

File: packages/fleet-agent/generated-changes/patch/Chart.yaml.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+--- charts-original/Chart.yaml
++++ charts/Chart.yaml
+@@ -5,8 +5,8 @@
+   catalog.cattle.io/os: linux
+   catalog.cattle.io/release-name: fleet-agent
+ apiVersion: v2
+-appVersion: 0.3.4-rc3
++appVersion: 0.3.3
```
Comment (thedadams) [inline packages/fleet-agent/generated-changes/patch/Chart.yaml.patch:8]: I think these versions should still be 0.3.4. When the patch is applied, I believe you want it to be 0.3.400-rc01

**Review by thedadams — CHANGES_REQUESTED** [review]

File: packages/fleet-agent/generated-changes/patch/Chart.yaml.patch (hunk 2 above)
Comment (nickgerace) [reply packages/fleet-agent/generated-changes/patch/Chart.yaml.patch]: Good catch. Strange that they are flipped. Fixed now

File: packages/fleet/generated-changes/patch/Chart.yaml.patch (hunk 1 above)
Comment (nickgerace) [reply packages/fleet/generated-changes/patch/Chart.yaml.patch]: Yep. Strange that they were flipped. Fixed now

File: packages/fleet-agent/generated-changes/patch/Chart.yaml.patch (hunk 2 above)
Comment (thedadams) [reply packages/fleet-agent/generated-changes/patch/Chart.yaml.patch]: You still need to remove the `-rc3`. The version should just be `0.3.4` and the rc will be added by Arvind's scripts.

**Review by thedadams — APPROVED** [review]

**Comment by nickgerace** [comment]
Investigating failing CI... This follows https://github.com/rancher/charts/pull/973#issuecomment-773444259, but there still may be an issue.

**Review by thedadams — APPROVED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Review by cbron — APPROVED** [review]

**Comment by StrongMonkey** [comment]
https://github.com/rancher/fleet/issues/239

**Comment by StrongMonkey** [comment]
https://github.com/rancher/fleet/issues/198

**Comment by StrongMonkey** [comment]
https://github.com/rancher/fleet/issues/232

**Pull request merged**